demos/demo_par_decimator/src/main.xc

```diff
@@ -0,0 +1,192 @@
+// Copyright 2023 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+
+#include "app_config.h"
+#include "util/mips.h"
+#include "device_pll_ctrl.h"
+
+#include "app.h"
+#include "util/audio_buffer.h"
+
+#include <platform.h>
+#include <xs1.h>
+#include <xclib.h>
+#include <xscope.h>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include <assert.h>
+
+unsafe{
```

Comment:
Obviously, it's convenient to place the entire main function in an unsafe scope.  However, for someone with less than an expert's understanding of XC, I believe it will be more helpful to only use unsafe scope where it's actually necessary.  In this function, that appears to be line 61 and no where else.